src/charm.py

```diff
@@ -59,7 +59,19 @@ def _certificate_validity(self) -> int:
         Returns:
             int: Certificate validity (in days)
         """
-        return int(self.model.config.get("certificate-validity", 365))
+        certificate_validity = int(self.model.config.get("certificate-validity", 365))
+        if certificate_validity > self._ca_certificate_validity:
```

Comment:
We should probably have this validation sooner. We already have a validation in the `config_changed` hook for `if not self._config_ca_common_name`. I'd simply add this if right after. And instead of simply logging, we can go to blocked state. It makes no sense to allow for certificate validity that is longer than the CA validity.

Reply:
Yes, that's totally fine.